plugins/usbdmx/AsyncPluginImpl.cpp

```diff
@@ -290,66 +321,63 @@ bool AsyncPluginImpl::USBDeviceAdded(libusb_device *usb_device) {
             << strings::ToHex(descriptor.idVendor) << ", product "
             << strings::ToHex(descriptor.idProduct);
 
-  WidgetFactories::iterator iter = m_widget_factories.begin();
-  for (; iter != m_widget_factories.end(); ++iter) {
-    if ((*iter)->DeviceAdded(&m_widget_observer, usb_device, descriptor)) {
-      STLReplacePtr(&m_device_factory_map, usb_device, *iter);
-      return true;
+  WidgetFactories::iterator factory_iter = m_widget_factories.begin();
+  for (; factory_iter != m_widget_factories.end(); ++factory_iter) {
+    if ((*factory_iter)->DeviceAdded(&m_widget_observer, usb_device,
+                                     descriptor)) {
+      state->factory = *factory_iter;
+      return;
     }
   }
-  return false;
-}
-
-/**
- * @brief Signal a USB device has been removed.
- *
- * This can be called from either the libusb thread or the main thread. However
- * only one of those will be active at once, so we can avoid locking.
- */
-void AsyncPluginImpl::USBDeviceRemoved(libusb_device *usb_device) {
-  WidgetFactory *factory = STLLookupAndRemovePtr(
-      &m_device_factory_map, usb_device);
-  if (factory) {
-    factory->DeviceRemoved(&m_widget_observer, usb_device);
-  }
 }
 
 /*
  * @brief Signal widget / device addition.
- * @param widget The widget that was added.
+ * @param device_id The device id to add.
```

Comment:
The type appears in the doxygen anyway. People reading the code can see the type in the signature.